### PR TITLE
feat(scan): mobile data layer for /scan/lookup endpoint (#697)

### PR DESCRIPTION
**Pull request by benoit-bremaud**
## Summary

Mobile counterpart of backend PR #729 / Issue #696. Closes the data path from the scan flow down to the new `GET /scan/lookup/:ean` endpoint while keeping the demo flow working offline.

Layered per Clean Architecture (`packages/mobile-app/CLAUDE.md`):

- **domain** — `ScanCatalogItem`, `ScanLookupResult`, `ScanLookupSource`, `ScanCatalogItemOrigin` added to `scan.types.ts` (camelCase domain shape).
- **data** — `scan-lookup.api.ts` calls `GET /scan/lookup/:ean` via the shared `http-client`, maps snake_case DTO -> camelCase, URL-encodes the EAN against path-traversal.
- **application** — `lookupBeerByBarcode` use-case:
  - normalises barcode (trim + strip non-digits) and rejects empty input with `ScanLookupInvalidBarcodeError` before any network call;
  - branches on `dataSource.useDemoData` and serves the on-device `demoScanCatalog` (Punk IPA, La Chouffe, Rochefort 10) so the soutenance demo works without network;
  - maps `HttpError 404 -> ScanLookupBeerNotFoundError` and `503 -> ScanLookupServiceUnavailableError`; other statuses bubble up unchanged for the presentation layer (#698).

## Tests (happy / sad / edge — 18 new, all green)

- 5 on data layer: URL building, snake -> camel mapping (filled), null fields (sad), URL-encoding for path-traversal (edge), HTTP error re-throw (edge).
- 13 on use-case layer: input normalisation (3), demo mode (2), HttpError mapping (5), custom error classes (3).

Full mobile suite: **439 passed / 52 suites**.

## Side fix — `.gitignore`

The unanchored `data/` rule (intended for the API SQLite folder) was silently ignoring the mobile Clean Architecture `data/` layer folder, so every new file under `packages/mobile-app/src/features/<feature>/data/` would be ignored by `git add`. Anchored to `/packages/api/data/` — backend behaviour unchanged, but new mobile data-layer files are no longer hidden.

## Deltas vs Issue #697 spec (worth flagging)

| Spec says | Shipped | Why |
| --- | --- | --- |
| `scan.api.ts` with `getBeerByBarcode` + `getRecipesForBeer` | `scan-lookup.api.ts` with `lookupBeerByBarcode` | Backend (#696/#729) ships a single `GET /scan/lookup/:ean` endpoint; the recipe-matching half is a separate vertical owned by #699. Function name mirrors the backend method 1:1. |
| `domain/beer-data.types.ts` + `domain/errors.ts` as new files | New types appended to `domain/scan.types.ts`; error classes co-located with the use-case | Keeps a single domain file per feature (existing convention) and the errors stay where they're thrown. Happy to split if the reviewer prefers. |

## CI

- `npm run ci:check` — lint, typecheck, format:check all green
- `npm test` — 439/439 (52 suites)

## Checklist

- [x] Happy path + sad path + edge cases covered
- [x] `tsc --noEmit` passes
- [x] Existing tests still green (439/439)
- [x] No `any`, no inline styles, named exports only
- [x] Respects `dataSource.useDemoData` toggle
- [x] No direct `fetch()` — uses `src/core/http/http-client`

Refs #697
Refs #594

### CODE REVIEW

File: packages/mobile-app/src/features/scan/application/scan-lookup.use-cases.ts [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+ *         failure (401, 429, 500, etc.) — caller decides how to
+ *         react.
+ */
+export async function lookupBeerByBarcode(
+  rawBarcode: string,
+): Promise<ScanLookupResult> {
+  const barcode = normaliseBarcode(rawBarcode);
+  if (barcode.length === 0) {
+    throw new ScanLookupInvalidBarcodeError(rawBarcode);
+  }
```
[the remark this answers is not in the document]
Comment (benoit-bremaud) [reply packages/mobile-app/src/features/scan/application/scan-lookup.use-cases.ts]: Good catch — agreed, this was a Should Have. Fixed in 5cb4353: added `BARCODE_MIN_LENGTH = 8` / `BARCODE_MAX_LENGTH = 14` constants mirroring the backend `^\d{8,14}$` contract from `ScanDomainService.validateBarcode`. The empty-string branch is now naturally subsumed by the `< 8` check, and 4 new unit tests cover the boundaries (too-short / too-long / 8-digit lower bound / 14-digit upper bound). No more useless network round-trip on malformed input.

File: packages/mobile-app/src/mocks/demo-data.ts [hunk 2]
```diff
@@ -2479,3 +2483,84 @@ export const demoBatches: Batch[] = [
...
+  abv: overrides.abv ?? null,
+  ibu: overrides.ibu ?? null,
+  colorEbc: overrides.colorEbc ?? null,
+  fermentationType: overrides.fermentationType ?? "ale",
+  aromaticTags: overrides.aromaticTags ?? null,
+  notesSource: overrides.notesSource ?? "demo data (mobile-app)",
+  isAbvEstimated: overrides.isAbvEstimated ?? false,
+  isIbuEstimated: overrides.isIbuEstimated ?? true,
+  isColorEbcEstimated: overrides.isColorEbcEstimated ?? true,
+  isStyleEstimated: overrides.isStyleEstimated ?? false,
```
[the remark this answers is not in the document]
Comment (benoit-bremaud) [reply packages/mobile-app/src/mocks/demo-data.ts]: Agreed — the defaults were misleading. Fixed in 5cb4353:

- `isAbvEstimated` / `isIbuEstimated` / `isColorEbcEstimated` now derive from whether the value is provided: `overrides.is*Estimated ?? overrides.<value> == null`. So a demo entry with a numeric IBU is no longer flagged as estimated.
- `notesSource` defaults to `null` (a truthful "no attribution" beats a fake "demo data" string).
- La Chouffe and Rochefort 10 entries now carry per-item `notesSource` strings tied to their real upstream documentation, matching the seed convention.

**Comment by benoit-bremaud** [comment]
## Fin de revue — PR #731

### CI

Toutes les checks vertes sur le commit fix `5cb4353` :
- ✅ `mobile-app` (lint + typecheck + tests + coverage) — 2m16s
- ✅ `changes` — 5s
- ✅ `security-audit` — 13s
- ✅ `SonarCloud Code Analysis` — 52s
- ✅ `SonarCloud Scan` — 3m30s
- ⏭️ `api` / `website` / `beer-encyclopedia` — skipping (path-filtered, normal — la PR ne touche que mobile-app + .gitignore)

### Reviews adressées (2/2)

| # | Reviewer | Sévérité | Sujet | Commit fix |
|---|----------|----------|-------|------------|
| 1 | Codex | P2 (Should Have) | Validate barcode length (8–14 digits) before lookup, mirror backend `^\d{8,14}$` contract | `5cb4353` |
| 2 | Copilot | Should Have | `buildDemoScanCatalogItem` defaults trompeurs (`is*Estimated: true` + `notesSource` générique) | `5cb4353` |

Détail des fixes :
- **#1** — ajout de `BARCODE_MIN_LENGTH = 8` / `BARCODE_MAX_LENGTH = 14`, le check "empty" est subsumé par le `< 8`. 4 nouveaux tests sur les bornes (too-short / too-long / 8-digit / 14-digit). Plus de round-trip réseau inutile sur input malformé.
- **#2** — `is*Estimated` dérive maintenant de la présence de la valeur (`overrides.is*Estimated ?? overrides.<value> == null`), `notesSource` default à `null`, La Chouffe et Rochefort 10 ont des `notesSource` per-item alignés avec leur documentation upstream.

### Tests

Suite complète mobile-app : **443 passed / 52 suites** (+4 nouveaux tests de bornes).

### Commits sur la PR

- `670862c` — chore(ci): trigger CI run after Actions budget unblock
- `5cb4353` — fix(scan): address Codex P2 + Copilot review on PR #731

### Prêt à merger ✅

Plus de commentaire ouvert. PR prête pour le merge manuel depuis l'UI GitHub.
